src/components/picker/index.tsx

```diff
@@ -89,7 +118,42 @@ export class SmoothlyPicker implements Clearable, Input {
 	async clear() {
 		this.selected.forEach(option => option.selected && option.element.clickHandler())
 	}
-	render() {
+	@Method()
+	async listen(property: "changed", listener: (parent: Editable) => Promise<void>): Promise<void> {
+		this.listener[property] = listener
+		listener(this)
+	}
+	@Method()
+	async edit(editable: boolean): Promise<void> {
+		await this.reset()
+		this.readonly = !editable
+	}
+	@Method()
+	async reset(): Promise<void> {
+		const initialValueArray = Array.from(this.initialValue.values(), option => option.value)
+		this.selected.forEach(
+			option => !initialValueArray.includes(option.value) && option.selected && option.element.setSelected(false)
+		)
+		this.initialValue.forEach(option => option.element.setSelected(true))
+	}
+	@Method()
+	async setInitialValue(): Promise<void> {
+		this.initialValue = new Map(this.selected)
+		this.selectedChanged()
+	}
+	areValuesEqual(selected: Map<any, Option>, initialValue: Map<any, Option>): boolean {
+		const initialValueArray = Array.from(initialValue.values(), option => option.value)
+		const selectedArray = Array.from(selected.values(), option => option.value)
+		if (selectedArray.length !== initialValueArray.length) {
+			return false
```

Comment:
We generally do not allow multiple return in a single function. make a variable `let result: boolean` and store the result in it and only return result at the end of the function. If you utilize early ejects from the function open up an else block or talk with someone on your team for an alternative solution.